src/testUtils.ts

```diff
@@ -89,10 +90,64 @@ export function getTestFunctions(doc: vscode.TextDocument, token: vscode.Cancell
 		.then(symbols =>
 			symbols.filter(sym =>
 				sym.kind === vscode.SymbolKind.Function
-				&& (sym.name.startsWith('Test') || sym.name.startsWith('Example')))
+				&& (sym.name.startsWith('Test') || sym.name.startsWith('Example') || testSuiteMethodRegex.test(sym.name))
+			)
 		);
 }
 
+/**
+ * Extracts test method name of a suite test function.
+ * For example a symbol with name "(*testSuite).TestMethod" will return "TestMethod".
+ *
+ * @param symbolName Symbol Name to extract method name from.
+ */
+export function extractInstanceTestName(symbolName: string): string {
+	const match = symbolName.match(testSuiteMethodRegex);
+	if (!match || match.length !== 3) {
+		return null;
+	}
+	return match[2];
+}
+
+/**
+ * Finds test method containing suite.Run() call for the given instance test name.
+ *
+ * @param symbolName Instance test function name (including the type name)
+ * @param doc Editor document
+ * @param allTests All test functions
+ */
+export function findTestFnForInstanceTest(symbolName: string, doc: vscode.TextDocument, allTests: vscode.SymbolInformation[]): vscode.SymbolInformation {
+	// get non-instance test functions
+	const testFunctions = allTests.filter(t => !testSuiteMethodRegex.test(t.name));
+	// filter further to ones containing suite.Run()
+	const candidates = testFunctions.filter(t => {
+		const text = doc.getText(t.location.range);
+		return /suite\.Run\(/.test(text);
+	});
+
+	switch (candidates.length) {
+		case 0: return null;
+		case 1: return candidates[0];
+	}
+
+	// we have multiple matches, filter even more
+	const match = symbolName.match(testSuiteMethodRegex);
+	if (!match || match.length !== 3) {
+		return candidates[0];
+	}
+	// drop pointers, lowercase for more lenient matching
+	const instanceType = match[1].replace(/[*]/g, '').toLowerCase();
+	const filtered = candidates.filter(t => {
+		const text = doc.getText(t.location.range).toLowerCase();
+		if (text.includes(instanceType)) {
```

Comment:
This may give false positives if say there is a comment inside the function with another suite's name.
It can give false negatives if the suite was instantiated outside the function.

Why not return all the functions that have a `suite.Run()`?

Since we also pass `-testify.m`, the other suites will be a no-op right?

Reply:
Changed, clearly this isn't going to work in 100% of cases, but should be close enough.